pulsar-functions/utils/src/main/java/org/apache/pulsar/functions/utils/FunctionConfigUtils.java

```diff
@@ -308,11 +308,20 @@ public static FunctionDetails convert(FunctionConfig functionConfig, ExtractedFu
         // windowing related
         WindowConfig windowConfig = functionConfig.getWindowConfig();
         if (windowConfig != null) {
+            // Windows Function not support EFFECTIVELY_ONCE
+            if (functionConfig.getProcessingGuarantees() == FunctionConfig.ProcessingGuarantees.EFFECTIVELY_ONCE) {
+                throw new IllegalArgumentException(
+                        "Windows Function not support EFFECTIVELY_ONCE delivery semantics.");
+            } else {
+                // Override functionConfig.getProcessingGuarantees to MANUAL, and set windowsFunction is guarantees
+                windowConfig.setProcessingGuarantees(WindowConfig.ProcessingGuarantees
```

Comment:
`WindowConfig.ProcessingGuarantees` only support `ATMOST_ONCE` and `ATLEAST_ONCE`, but `functionDetails.ProcessingGuarantees` do have `MANUAL` and `EFFECTIVELY_ONCE` as well. This block do ignores `EFFECTIVELY_ONCE` case, but what if the `ProcessingGuarantees` is `MANUAL`? The `WindowConfig.ProcessingGuarantees.valueOf` may throw `IllegalArgumentException`, is it as expected? If yes, can we have some better logging here?

Reply:
Thanks, `WindowsFunction` does not support `MANUAL`, I will add some checks.